### PR TITLE
SOLR-13727: Bug fix for V2Request handling in HttpSolrClient

### PR DESCRIPTION
**Pull request by megancarey**
# Description

This change addresses a bug that was identified in the HttpSolrClients. Because the createMethod method just uses string.replace, it's possible that the wrong portion of the host address string is replaced. The intention is to change the start of the path from "/solr" to "/api", but if the hostname begins with "solr", it replaces the beginning of the hostname. 

For example, if the address is `http://solr-host.com:8983/solr/`, instead of changing to `http://solr-host.com:8983/api/` it will become `http:/api-host.com:8983/solr/`.

# Solution

My solution is to create a java.net URL object from the address string, replace the beginning of the path as expected, create a new URL object, and return the string value. This also ensures that the URL is properly formatted.

# Tests

There are no additional tests, but I have verified when running solr locally that this change fixes the bug.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I am authorized to contribute this code to the ASF and have removed any code I do not have a license to distribute.
- [x] I have developed this patch against the `master` branch.
- [x] I have run `ant precommit` and the appropriate test suite.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Ref Guide](https://github.com/apache/lucene-solr/tree/master/solr/solr-ref-guide) (for Solr changes only).

### CODE REVIEW

**Comment by yonik** [comment]
Hi Megan, I think this PR accidentally has changes from your previous PR as well.  Could you do one with just the described changes?

**Comment by megancarey** [comment]
@yonik The changed files contain only my changes for this bug fix, but does have other commits attached - would it be easier if I condensed the commit history to a single commit?

**Comment by yonik** [comment]
Hmmm, I do see "files changed 2", but I also see "Commits 7".  I guess I'm getting confused by the merge commits.  I wonder if using the "Squash and merge" from github will keep a linear history?  I think I'll be safe and just cherry-pick the last two commits.

File: solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpSolrClient.java [hunk 1]
```diff
@@ -330,6 +333,12 @@ protected ModifiableSolrParams calculateQueryParams(Set<String> queryParamNames,
     }
     return queryModParams;
   }
+  
+  private String changeV2RequestEndpoint(String basePath) throws MalformedURLException {
```
Comment (noblepaul) [inline solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpSolrClient.java]: this could be made a `public static` method

Comment (yonik) [reply solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpSolrClient.java]: This is a widely used client class, so we should probably be slightly more wary of adding public methods.  Perhaps package protected static... until we need it to be public for some reason.  I'll make that change when I commit.

**Comment by yonik** [comment]
Committed manually to master & 8x.  Thanks Megan!

**Pull request closed without merging**